pkg/istio-agent/xds_proxy.go

```diff
@@ -712,6 +714,7 @@ func (p *XdsProxy) tapRequest(req *discovery.DiscoveryRequest, timeout time.Dura
 
 	// Send to Istiod
 	p.connected.sendRequest(req)
+	p.connectedMutex.Unlock()
```

Comment:
Donot make the two locks intersected. You can first get the connection, then do the following process

	p.connectedMutex.Lock()
        connection := p.connected
	p.connectedMutex.Lock()